A41-xds-rbac.md

```diff
@@ -0,0 +1,199 @@
+XdsClient must NACK the configuration.
+
+The following fields of `Permission` may not be obvious how they map to gRPC:
+
+| Permission Field | gRPC Equivalent |
+| ---------------- | --------------- |
+| header           | Metadata (with caveats) |
+| url_path         | Fully-qualified RPC method name with leading slash |
+| destination_ip   | Local address for this connection |
+| destination_port | Local port for this connection |
+| metadata         | Hard-coded as empty; never matches |
+| requested_server_name | Hard-coded as empty string |
+
+Be strongly aware that Envoy Metadata has no relation to gRPC's Metadata.
+Because matching supports NOT, the matcher must still be processed even if a
+rule contains references to things that don't generally match; it is not trivial
+to "optimize out" the never-matching rules.
+
+The `header` field is not entirely 1:1 with gRPC Metadata. To begin with, gRPC
+Metadata isn't 100% consistent cross-language in its handling of hop-to-hop
+headers (e.g., `TE`) and pseudo-headers. For this design, `headers` can include
+`:method`, `:scheme`, `:authority`, and`:path` matchers and they must match the
+values received on-the-wire independent of whether they are stored in Metadata
+or in separate APIs. It is unspecified whether hop-to-hop headers are matched.
+Multi-valued metadata is represented as the concatenation of the values along
+with a `,` (comma, no added spaces) separator, as permitted by HTTP and gRPC.
+Binary headers are represented in their base64-encoded form, although we rarely
+expect binary header matchers.
+
+TODO: Clearly define differences compared to client-side. Align with client-side
+routing (maybe changing behavior) to allow reusing code, have a strong grip on
+client vs server differences, and notice 'gotcha's.  Content-Type is probably
+not a problem on server-side.
+
+As documented for `HeaderMatcher`, Envoy aliases `:authority` and `Host` in its
+header map implementation, so they should be treated equivalent for the RBAC
+matchers; there must be no behavior change depending on which of the two header
+names is used in the _RBAC policy_.
+
+TODO: We've had no reason to care about :scheme in our APIs and so is not
+present in all languages. It could be inferred from whether the connection used
+security, but that's technically different from :scheme, as if you go through a
+TLS-terminating proxy and use plaintext to the backend the :scheme is used to
+re-create the original URL which would be 'https', not 'http'. This may be a
+problem for Java and Go. We can maybe copy whatever Envoy does for HTTP/1
+traffic. You might think :scheme doesn't matter, and maybe it doesn't, but it
+might be used to detect "is the client using TLS". Although note that in that
+case such a check is trusting the client to provide the correct value.
+
+TODO: Need to audit that server implementations check that :method=POST before
```

Comment:
I thought this must be part of some conformance or interop tests because this is the basic [gRPC spec](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests). If not we should make it.